data/src/error_analysis/erroranalysis.py

```diff
@@ -0,0 +1,70 @@
+"""Error Analysis tool for G2P.
+
+This script assumes 2 input files. a) Covering Grammer b) Test output. 
+CG file  : Each line contain graphime and their corresponding pronunciation seperated by a tab
+Test File: contains three attributes in each line seperated by a tab. The attributes are - orthography, expected pronunciation, and hypothesized pronunciation."""
+
+__author__ = "Arundhati Sengupta"
+
+import pynini
+import argparse
+
+def main(args: argparse.Namespace):
+    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+    rulematch_predMatch = 0
+    rulematch_predNotMatch = 0
+    notRulematch_predMatch = 0
+    notRulematch_predNotMatch = 0
+    total_records = 0
+    with open(args.test_path, "r") as source:
+        for line in source:
+            total_records+=1
+            parts = line.split('\t')
+            lg = parts[0].strip()
+            act = parts[1].replace(' ','').replace('.','').strip()
+            predPron = parts[2].replace(' ','').replace('.','').strip()
+             
```

Comment:
The blank lines within this function strikes me as ad-hoc. I'd recommend getting rid of it altogether, I think it makes the function harder to read.

Reply:
fixed.